engine/consensus/compliance/core.go

```diff
@@ -225,26 +252,36 @@ func (c *Core) OnBlockProposal(originID flow.Identifier, proposal *messages.Bloc
 // its pending proposals for its children. By induction, any children connected
 // to a valid proposal are validly connected to the finalized state and can be
 // processed as well.
-func (c *Core) processBlockAndDescendants(proposal *messages.BlockProposal, inRangeBlockResponse bool) error {
+// No errors are expected during normal operation. All returned exceptions
+// are potential symptoms of internal state corruption and should be fatal.
+func (c *Core) processBlockAndDescendants(proposal *messages.BlockProposal, parent *flow.Header) error {
 	blockID := proposal.Header.ID()
 
+	log := c.log.With().
+		Str("block_id", blockID.String()).
+		Uint64("block_height", proposal.Header.Height).
+		Uint64("block_view", proposal.Header.View).
+		Uint64("parent_view", parent.View).
+		Logger()
+
 	// process block itself
-	err := c.processBlockProposal(proposal, inRangeBlockResponse)
-	// child is outdated by the time we started processing it
-	// => node was probably behind and is catching up. Log as warning
-	if engine.IsOutdatedInputError(err) {
-		c.log.Info().Msg("dropped processing of abandoned fork; this might be an indicator that the node is slightly behind")
-		return nil
-	}
-	// the block is invalid; log as error as we desire honest participation
-	// ToDo: potential slashing
-	if engine.IsInvalidInputError(err) {
-		c.log.Warn().
-			Err(err).
-			Msg("received invalid block from other node (potential slashing evidence?)")
-		return nil
-	}
+	err := c.processBlockProposal(proposal, parent)
 	if err != nil {
+		if checkForAndLogOutdatedInputError(err, log) {
+			return nil
+		}
+		if checkForAndLogInvalidInputError(err, log) || checkForAndLogUnverifiableInputError(err, log) {
```

Comment:
```suggestion
		if checkForAndLogInvalidInputError(err, log) {
```

Suggesting to modify the error types used here to be consistent with how they are used in cluster consensus. In particular, use invalid input only for cases where we know the entity to be invalid - use unverifiable input when we're not sure.